tests/scripts/thread-cert/border_router/test_multi_thread_networks.py

```diff
@@ -130,17 +130,10 @@ def test(self):
 
         # Each BR should independently register an external route for the on-link prefix
         # and OMR prefix in another Thread Network.
-        self.assertTrue(len(br1.get_netdata_non_nat64_prefixes()) == 1)
-        self.assertTrue(len(router1.get_netdata_non_nat64_prefixes()) == 1)
-        self.assertTrue(len(br2.get_netdata_non_nat64_prefixes()) == 1)
-        self.assertTrue(len(router2.get_netdata_non_nat64_prefixes()) == 1)
-
-        br1_external_routes = br1.get_routes()
-        br2_external_routes = br2.get_routes()
-
-        br1_external_routes.sort()
-        br2_external_routes.sort()
-        self.assertNotEqual(br1_external_routes, br2_external_routes)
```

Comment:
The earlier check always passes when BR adds /64 external routes but it fails occasionally when BR adds the generic route (`fc00::/7`).

1. When BR added specific /64 external routes, the external route lists are always different (each BR has the other's OMR prefix as an external route) so the check always succeeds.
2. When BR added generic external routes, the `br1_external_routes` and `br2_external_routes` will only have 1 entry `fc00::/7 sa med <rloc16>`. Most of times the check will still pass because BR1 and BR2 are assigned with different RLOC16s. However, BR1 and BR2 are in different networks so it's possible that they have the same RLOC16. In such a case the check will fail. 

I noticed a CI run that the check failed because BR1 and BR2 had the same RLOC16 so I sent this PR to fix it.